src/ripple/nodestore/impl/DatabaseShardImp.h

```diff
@@ -232,7 +232,7 @@ class DatabaseShardImp : public DatabaseShard
     std::uint32_t ledgersPerShard_ = ledgersPerShardDefault;
 
     // The earliest shard index
-    std::uint32_t const earliestShardIndex_;
+    std::uint32_t earliestShardIndex_;
```

Comment:
It looks like you removed the `const` qualifier in order to set `earliestShardIndex_` in `DatabaseShardImp::initConfig`. Why is that necessary when this variable gets initialized during the constructor?

Reply:
See previous comment.

Reply:
I understand why you did this - it's a consequence of the two phase initialization and testing. I'm OK with the change, but I'd add a comment explaining why this is not const.